metric_namer.go

```diff
@@ -120,7 +120,10 @@ func (mn *MetricNamer) buildCompliantMetricName(name, unit string, metricType Me
 
 	// Namespace?
 	if mn.Namespace != "" {
-		return mn.Namespace + "_" + metricName
+		namespace := strings.Join(strings.FieldsFunc(mn.Namespace, func(r rune) bool {
+			return invalidMetricCharRE.MatchString(string(r))
+		}), "_")
```

Comment:
This seems rather expensive.

Reply:
Note that function already exists `isValidCompliantMetricChar`.  
It's missing `_` as a valid character, which I don't understand.